tensorflow_addons/image/cutout_ops.py

```diff
@@ -0,0 +1,166 @@
+    constant_values: Number = 0,
+    seed: Number = None,
+    data_format: str = "channels_last",
+) -> tf.Tensor:
+    """Apply cutout (https://arxiv.org/abs/1708.04552) to images.
+
+    This operation applies a (2 * mask_height x 2 * mask_width) mask of zeros to
+    a random location within `img`. The pixel values filled in will be of the
+    value `replace`. The located where the mask will be applied is randomly
+    chosen uniformly over the whole images.
+
+    Args:
+      images: A tensor of shape
+        (batch_size, height, width, channels)
+        (NHWC), (batch_size, channels, height, width)(NCHW).
+      mask_size: Specifies how big the zero mask that will be generated is that
+        is applied to the images. The mask will be of size
+        (2 * mask_height x 2 * mask_width).
+      constant_values: What pixel value to fill in the images in the area that has
+        the cutout mask applied to it.
+      seed: A Python integer. Used in combination with `tf.random.set_seed` to
+        create a reproducible sequence of tensors across multiple calls.
+      data_format: A string, one of `channels_last` (default) or `channels_first`.
+        The ordering of the dimensions in the inputs.
+        `channels_last` corresponds to inputs with shape
+        `(batch_size, ..., channels)` while `channels_first` corresponds to
+        inputs with shape `(batch_size, channels, ...)`.
+    Returns:
+      An image Tensor.
+    """
+    batch_size = tf.shape(images)[0]
+    mask_size, data_format, image_height, image_width = _norm_params(
+        images, mask_size, data_format
+    )
+
+    cutout_center_height = tf.random.uniform(
+        shape=[batch_size], minval=0, maxval=image_height, dtype=tf.int32, seed=seed
+    )
+    cutout_center_width = tf.random.uniform(
+        shape=[batch_size], minval=0, maxval=image_width, dtype=tf.int32, seed=seed
+    )
+
+    offset = tf.transpose([cutout_center_height, cutout_center_width], [1, 0])
+    return cutout(images, mask_size, offset, constant_values, data_format,)
+
+
+def cutout(
+    images: TensorLike,
+    mask_size: TensorLike,
+    offset: TensorLike = (0, 0),
+    constant_values: Number = 0,
+    data_format: str = "channels_last",
+) -> tf.Tensor:
+    """Apply cutout (https://arxiv.org/abs/1708.04552) to images.
+
+    This operation applies a (2 * mask_height x 2 * mask_width) mask of zeros to
+    a random location within `img`. The pixel values filled in will be of the
+    value `replace`. The located where the mask will be applied is randomly
+    chosen uniformly over the whole images.
+
+    Args:
+      images: A tensor of shape (batch_size, height, width, channels)
+        (NHWC), (batch_size, channels, height, width)(NCHW).
+      mask_size: Specifies how big the zero mask that will be generated is that
+        is applied to the images. The mask will be of size
+        (2 * mask_height x 2 * mask_width).
+      offset: A tuple of (height, width)
```

Comment:
Could you specify all the shapes are are possible? A tuple is possible but a 2D tensor too (batch_size, 2)